src/lib/notifier.c

```diff
@@ -54,9 +53,7 @@ int notifier_register(void *receiver, void *caller, enum notify_id type,
 	handle->caller = caller;
 	handle->cb = cb;
 
-	spin_lock(notify->lock);
```

Comment:
I assume notifier_register will never be called in interrupt context and process context as well. But I guess that would have lead to deadlocks.

I feel like there is more to this hence when I was trying to debug my crashes within notifier code last month (which weren't actually related to notifier) I did look at these as well.